Geek-Collectors-Network/geek-collectors-network
# 90 create user profile card UI component

### Summary
Created a simple `UserProfileCard` component. The goal is to create something reusable. To achieve this, the component has three props: `name`, `imageURL`, and `icons`. 

The `name` and `imageURL` allow us to render different user names and images; the `icon` allows us to provide different icons depending on the context of the component. 

For example, if the user is looking at a friend list, the icons will be a "mail" and "social media" icon; if it's a pending request, the icons will be "check" (accept) or "x" (decline). We can then define the `onClick` functionality for those icons within the context (the page) where they're used.

See the `TestPage` for how I've used the component, including how I've created and passed in the icons, name, and imageURL.

** While waiting on this, I'll experiment with creating pages that render a series of `UserProfileCard`s; this will include `useState` and `useEffect` **

### Reviewer's Notes
- We'll need to include a generic profile photo if, for whatever reason, the user hasn't uploaded one or it has problems rendering.
- The component is designed for mobile; it works for desktop, and there's basic dynamic rendering to provide a larger image and text, but this could be improved.
- I've set a minimum width of 375 pixels for the component; this is a hardcoded value, obviously, and we may want to change this, but I doubt it'll get much smaller than this (the size of an iPhone SE, one of the smaller smartphones on the markets). 

### Testing
I haven't added a route to the test page, so using Deploy Preview won't work. To see how the component looks, do the following:

1. Navigate to `frontend`
2. Run `npm run start:dev`
3. Open a browser at `http://localhost:8080/test`
4. Check out the component in mobile mode
